src/linker/Linker.Steps/BaseStep.cs

```diff
@@ -35,21 +35,19 @@ namespace Mono.Linker.Steps
 	public abstract class BaseStep : IStep
 	{
 
-		private LinkContext _context;
+		private LinkContext? _context;
 
-		public LinkContext Context {
-			get { return _context; }
-		}
+		public LinkContext Context => _context ?? throw new InvalidOperationException ();
```

Comment:
I don't like this patter where your turn a simple optimizable property into something that is complex check for unnecessary condition.

Reply:
Is the concern about the performance impact of the extra check? One alternative would be to return `_context!` and leave this as a nullref exception if accessed incorrectly. From an API design standpoint I'd rather throw `InvalidOperationException` if the performance is acceptable. Let me know what you prefer or if you have any other suggestions!